src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -135,4 +173,36 @@ private static void remove(Block b) {
             hologram.remove();
         }
     }
+
+    private void createSelector(int menuSlot, ItemStack button, char coordinate, char direction, String parameter, float smallChange, float largeChange, Block projector, ChestMenu menu) {
+
+        menu.addItem(menuSlot, new CustomItem(button,
+            "&7" + coordinate + " Offset (" + direction + ") &e" + DoubleHandler.fixDouble(Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), parameter))),
+            "", "&eLeft Click: &7+0.1", "&eRight Click: &7+0.5"));
+        menu.addMenuClickHandler(menuSlot, (pl, slot, item, action) -> {
+            double storedOffset = Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), parameter));
+            double offset;
+
+            offset = action.isRightClicked() ? largeChange : smallChange;
+
+            if ((coordinate == 'X' || coordinate == 'Z')
+                && offset + storedOffset > maxDistance.getValue()) {
+                return false;
+            }
+            ArmorStand hologram = getArmorStand(projector, true);
+
+            if (coordinate == 'X') {
+                hologram.teleport(hologram.getLocation().add(offset, 0, 0));
+            } else if (coordinate == 'Y') {
+                hologram.teleport(hologram.getLocation().add(0, offset, 0));
+            } else if (coordinate == 'Z') {
+                hologram.teleport(hologram.getLocation().add(0, 0, offset));
+            }
```

Comment:
```
hologram.teleport(hologram.getLocation().add(xOffset, yOffset, zOffset);
```

Make the offsets parameters, and remove these parameters:
coordinate, direction, smallChange, largeChange

Reply:
x and z are handled differently than y, wouldn't giving them all their own variables me less efficient?

Reply:
Well not sure tbh I just thought it would look  bit better. @TheBusyBiscuit what do you think?

Reply:
I think it would be fine like that, but I am suggesting an enum for this anyway, so you could move this into the enum as a method anyway.